schemas/tls.yml

```diff
@@ -0,0 +1,174 @@
+---
+- name: tls
+  title: TLS
+  group: 2
+  short: Fields describing a TLS connection.
+  description: >
+    Fields related to TLS activity.
+  type: group
+  fields:
+
+    - name: version.protocol
+      level: extended
+      type: keyword
+      description: Normalized protocol name parsed from original string (e.g. ssl, tls).
+
+    - name: version.number
+      level: extended
+      type: keyword
+      description: Numeric part of the version parsed from the original string (e.g. 1.2, 3).
+
+    - name: cipher
+      type: keyword
+      level: extended
+      description: String indicating the cipher used during the current connection.
+
+    - name: curve
+      type: keyword
+      level: extended
+      description: String indicating the curve used for the given cipher, when applicable.
+
+    - name: resumed
+      type: boolean
+      level: extended
+      description: Boolean flag indicating if this TLS connection was resumed from an existing TLS negotiation.
+
+    - name: next_protocol
+      type: keyword
+      level: extended
+      description: String indicating the protocol being tunneled (e.g. http/1.1, spdy/3, imap, webrtc)
+
+    - name: client.ja3
```

Comment:
I'm not an expert on the subject, but is it intentional that the 2 JA3 fields aren't exactly named the same? Currently we have:

`tls.client.ja3`
`tls.server.ja3s`

Just wondering about the trailing "s"

Reply:
Another question around consistency: isn't JA3 a hash as well? Shouldn't we add it at `*.hash.ja3`?

On the other hand I do get the sense JA3 is especially powerful/relevant for TLS, and if we want to make it more of a first class field this way, perhaps that's ok as well

Reply:
JA3 is specific to TLS/SSL. There are some similar hash schemes for things like SSH and others. Both fingerprints are ultimately MD5s, but they are hashing a specific thing.

For `ja3`, the following fields are hashed:

```SSLVersion,Cipher,SSLExtension,EllipticCurve,EllipticCurvePointFormat```

For `ja3s`, the following fields are hashed:

```SSLVersion,Cipher,SSLExtension```

Reply:
Got it so ja3 and ja3s it is, thanks for the clarification :-)